pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1442,3 +1442,44 @@ def test_merge_series(on, left_on, right_on, left_index, right_index, nm):
         with pytest.raises(ValueError, match='a Series without a name'):
             result = pd.merge(a, b, on=on, left_on=left_on, right_on=right_on,
                               left_index=left_index, right_index=right_index)
+
+
+@pytest.mark.parametrize("col1, col2, suffixes, expected_cols", [
+    (0, 0, ("", "_dup"), ["0", "0_dup"]),
+    (0, 0, (None, "_dup"), [0, "0_dup"]),
+    (0, 0, ("_x", "_y"), ["0_x", "0_y"]),
+    ("a", 0, (None, "_y"), ["a", 0]),
+    (0.0, 0.0, ("_x", None), ["0.0_x", 0.0]),
+    ("b", "b", (None, "_y"), ["b", "b_y"]),
+    ("a", "a", ("_x", None), ["a_x", "a"]),
+    ("a", "b", ("_x", None), ["a", "b"]),
+    ("a", "a", [None, "_x"], ["a", "a_x"]),
+    (0, 0, ["_a", None], ["0_a", 0])
```

Comment:
for added assurance can you also add a couple of regression tests here
```
("a", "a", None, ["a_x", "a_y"]),
(0, 0, None, ["0_x", "0_y"])
```
should be sufficient.

Reply:
you mean `("a", "a", ("_x", "_y"), ["a_x", "a_y"])` probably? suffix cannot accept `None` for now. @simonjayhawkins

Reply:
thanks! just added a small new test function to cover it!